crates/matrix-sdk/src/test_utils/mocks/mod.rs

```diff
@@ -3243,6 +3243,244 @@ impl<'a> MockEndpoint<'a, LoginEndpoint> {
     }
 }
 
+/// A struct for building a login response with status code 200
+#[derive(Default)]
+pub struct LoginResponseBuilder200 {
+    access_token: Option<String>, // required
+    device_id: Option<String>,   // required
+    expires_in_ms: Option<u32>,
+    refresh_token: Option<String>,
+    user_id: Option<String>,    // required
+    homeserver_url: Option<String>, // required if well_known used
+    identity_url: Option<String>, // required if well_known and m.identity_server used
+}
+
+impl LoginResponseBuilder200 {
+    /// constructor for empty reponse 
+    pub fn new() -> Self { Self::default() }
+
+    /// sets access_token
+    pub fn set_access_token(&mut self, value: &str) -> &mut Self { 
```

Comment:
Our naming convention avoids the usage of `set`, it's obvious that those will be setters.